### PR TITLE
fix(components): remove some proptypes warning output

### PR DESCRIPTION
**Pull request by jmfrancois**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our
  [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)

The testU and the docs output the following warning:

```Warning: Failed prop type: Invalid prop `data` of value `[object Object],[object Object]` supplied to `Table`, expected one of [null,null,null,null,null].```

```Warning: Failed prop type: Invalid prop `edit` of type `function` supplied to `NativeValue`, expected `boolean`.```

```Warning: Failed prop type: Invalid prop `label` of type `number` supplied to `TooltipTrigger`, expected `string`.```

```
Warning: Each child in an array or iterator should have a unique "key" prop. Check the render method of `Action`. See https://fb.me/react-warning-keys for more information.
          in span (created by Action)
          in Action (created by ListHeaders)
          in th (created by ListHeaders)
          in tr (created by ListHeaders)
          in ListHeaders (created by DisplayTable)
          in thead (created by DisplayTable)
          in table (created by DisplayTable)
          in div (created by DisplayTable)
          in div (created by DisplayTable)
          in DisplayTable (created by Unknown)
          in Unknown
```

```
Warning: Failed prop type: The prop `name` is marked as required in `Icon`, but its value is `undefined`.
          in Icon
```

```
    console.error node_modules/fbjs/lib/warning.js:36
      Warning: Failed prop type: The prop `id` is required to make `Dropdown` accessible for users of assistive technologies such as screen readers.
          in Dropdown (created by Uncontrolled(Dropdown))
          in Uncontrolled(Dropdown) (created by SplitButton)
          in SplitButton (created by ActionSplitDropdown)
          in ActionSplitDropdown
```

```
      Warning: Failed prop type: The prop `onClick` is marked as required in `Action`, but its value is `undefined`.
          in Action (created by SidePanel)
          in SidePanel (created by Unknown)
          in Unknown
```

**What is the new behavior?**

No more proptypes warning on ```jest -t Table```

screen shot 2017-03-21 at 11 01 21

Some are still here:

``` PASS  src/List/Toolbar/Filter/Filter.snapshot.test.js
  ● Console

    console.error node_modules/fbjs/lib/warning.js:36
      Warning: Failed prop type: The prop `onFilter` is marked as required in `Filter`, but its value is `undefined`.
          in Filter
    console.error node_modules/fbjs/lib/warning.js:36
      Warning: Failed prop type: The prop `onToggle` is marked as required in `Filter`, but its value is `undefined`.
          in Filter
    console.error node_modules/fbjs/lib/warning.js:36
      Warning: Failed prop type: The prop `onFilter` is marked as required in `FilterInput`, but its value is `undefined`.
          in FilterInput (created by Filter)
          in Filter
    console.error node_modules/fbjs/lib/warning.js:36
      Warning: Failed prop type: The prop `onClick` is marked as required in `Action`, but its value is `undefined`.
          in Action (created by Filter)
          in Filter
```

```
 PASS  src/AppHeaderBar/AppHeaderBar.snapshot.test.js
  ● Console

    console.info node_modules/@kadira/storybook/dist/server/babel_config.js:69
      => Loading custom .babelrc
    console.error node_modules/fbjs/lib/warning.js:36
      Warning: Failed prop type: Invalid prop `id` of type `number` supplied to `Typeahead`, expected `string`.
          in Typeahead (created by AppHeaderBar)
          in AppHeaderBar
          in div (created by Story)
          in div (created by Story)
          in Story
          in div
    console.error node_modules/fbjs/lib/warning.js:36
      Warning: Failed prop type: Invalid prop `id` of type `number` supplied to `Autowhatever`, expected `string`.
          in Autowhatever (created by Typeahead)
          in Typeahead (created by AppHeaderBar)
          in div (created by NavbarForm)
          in NavbarForm (created by AppHeaderBar)
          in div (created by NavbarCollapse)
          in Transition (created by Collapse)
          in Collapse (created by NavbarCollapse)
          in NavbarCollapse (created by AppHeaderBar)
          in div (created by Grid)
          in Grid (created by Navbar)
          in nav (created by Navbar)
          in Navbar (created by Uncontrolled(Navbar))
          in Uncontrolled(Navbar) (created by AppHeaderBar)
          in AppHeaderBar
          in div (created by Story)
          in div (created by Story)
          in Story
          in div
```

```
PASS  src/List/Toolbar/Toolbar.snapshot.test.js
  ● Console

    console.error node_modules/fbjs/lib/warning.js:36
      Warning: Failed prop type: The prop `display.onChange` is marked as required in `Toolbar`, but its value is `undefined`.
          in Toolbar
    console.error node_modules/fbjs/lib/warning.js:36
      Warning: Failed prop type: The prop `onChange` is marked as required in `SelectDisplayMode`, but its value is `undefined`.
          in SelectDisplayMode (created by Toolbar)
          in Toolbar
    console.error node_modules/fbjs/lib/warning.js:36
      Warning: Failed prop type: The prop `pagination.onChange` is marked as required in `Toolbar`, but its value is `undefined`.
          in Toolbar
    console.error node_modules/fbjs/lib/warning.js:36
      Warning: Failed prop type: The prop `onChange` is marked as required in `Pagination`, but its value is `undefined`.
          in Pagination (created by Toolbar)
          in Toolbar
```

```
 PASS  src/Drawer/Drawer.test.js
  ● Console

    console.error node_modules/fbjs/lib/warning.js:36
      Warning: Failed prop type: The prop `footerActions` is marked as required in `Drawer`, but its value is `undefined`.
          in Drawer
    console.error node_modules/fbjs/lib/warning.js:36
      Warning: Failed prop type: Drawer: prop type `onCancelAction` is invalid; it must be a function, usually from React.PropTypes.
          in Drawer
    console.error node_modules/fbjs/lib/warning.js:36
      Warning: Failed prop type: The prop `title` is marked as required in `DrawerTitle`, but its value is `undefined`.
          in DrawerTitle (created by Drawer)
          in Drawer
    console.error node_modules/fbjs/lib/warning.js:36
      Warning: Failed prop type: DrawerTitle: prop type `onCancelAction` is invalid; it must be a function, usually from React.PropTypes.
          in DrawerTitle (created by Drawer)
          in Drawer
    console.error node_modules/fbjs/lib/warning.js:36
      Warning: Failed prop type: DrawerTitle: prop type `children` is invalid; it must be a function, usually from React.PropTypes.
          in DrawerTitle (created by Drawer)
          in Drawer
    console.error node_modules/fbjs/lib/warning.js:36
      Warning: Failed prop type: Invalid prop `actions` of type `array` supplied to `ActionBar`, expected `object`.
          in ActionBar (created by Drawer)
          in Drawer
    console.error node_modules/fbjs/lib/warning.js:36
      Warning: Failed prop type: The prop `label` is marked as required in `Action`, but its value is `undefined`.
          in Action
    console.error node_modules/fbjs/lib/warning.js:36
      Warning: Failed prop type: The prop `onClick` is marked as required in `Action`, but its value is `undefined`.
          in Action
```

```
 PASS  src/ActionBar/ActionBar.snapshot.test.js
  ● Console

    console.error node_modules/fbjs/lib/warning.js:36
      Warning: Failed prop type: Invalid prop `actions.left[0]` supplied to `ActionBar`.
          in ActionBar```

**Does this PR introduce a breaking change?**

- [x] No

### CODE REVIEW

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://303.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://303.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://303.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://303.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://303.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://303.talend.surge.sh)

File: packages/components/src/List/DisplayTable/DisplayTable.component.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ function cellContent(isTitle, item, column, titleProps, id) {
 		/>);
 	}
 	return (
-		<TooltipTrigger label={item[column.key]} tooltipPlacement="top">
+		<TooltipTrigger label={`${item[column.key]}`} tooltipPlacement="top">
```
Comment (acateland) [inline packages/components/src/List/DisplayTable/DisplayTable.component.js]: add test showing that even if `item[column.key]` return undefined, label will be an empty string

File: packages/components/src/List/ItemTitle/ItemTitle.component.js [hunk 2]
```diff
@@ -146,7 +146,7 @@ function ItemTitle({ id, className, item, titleProps }) {
 	return (
 		<div className={theme['item-title']}>
 			{icon}
-			<TooltipTrigger label={value} tooltipPlacement="top">{titleElement}</TooltipTrigger>
+			<TooltipTrigger label={`${value}`} tooltipPlacement="top">{titleElement}</TooltipTrigger>
```
Comment (acateland) [inline packages/components/src/List/ItemTitle/ItemTitle.component.js]: add test showing that even if `value` is undefined, label will be an empty string

File: packages/components/src/List/DisplayTable/DisplayTable.test.js [hunk 3]
```diff
@@ -101,3 +106,74 @@ describe('DisplayTable', () => {
...
+		expect(wrapper.node).toMatchSnapshot();
+	});
+	it('should render undefined', () => {
+		const wrapper = shallow(<CellContent
+			column={{ key: 'test' }}
+			item={{}}
+		/>);
+		expect(wrapper.node).toMatchSnapshot();
+	});
+	it('should render array', () => {
```
Comment (acateland) [inline packages/components/src/List/DisplayTable/DisplayTable.test.js]: the snapshot doesn't display the label 
```
<TooltipTrigger
  label=""
  tooltipPlacement="top"
>
  <span
    className="item-text"
  />
</TooltipTrigger>
```

File: packages/components/src/List/ItemTitle/__snapshots__/ItemTitle.test.js.snap [hunk 4]
```diff
@@ -0,0 +1,115 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`ItemTitle should render button title 1`] = `
+<div
+  className={undefined}
```
Comment (acateland) [inline packages/components/src/List/ItemTitle/__snapshots__/ItemTitle.test.js.snap]: seems to be always render as undefined, should it be dropped ?

Comment (jmfrancois) [inline packages/components/src/List/ItemTitle/__snapshots__/ItemTitle.test.js.snap]: the className is undefined because we use theme['item-title'] sass and in tests it's mocked.

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://303.talend.surge.sh)
